client.cpp

```diff
@@ -27,20 +52,27 @@ int main(int argc, char *argv[]) {
     char *filepath = argv[2];
 
     // read in file 
-    char *data;
-    streampos data_len;
+    char data[MAX_DATA_SIZE];
+    long data_len;
```

Comment:
data_len really should stay as type streampos, as it's the most robust way of working with iostreams. A streampos can actually be larger than a long long, so has potential for causing problems if we choose to send large enough files (I believe 4G and above, but I would need to check on that).  Unlikely to affect us on this project, but best practice would be to keep it as streampos and smart cast it (conditional to check if it will fit in a long long before casting) to another variable if necessary to do so. :)